lib/eqaf.ml

```diff
@@ -183,4 +183,32 @@ let[@inline always] select_int choose_b a b =
   let mask = ((- choose_b) lor choose_b) asr Sys.int_size in
   (a land (lnot mask)) lor (b land mask)
 
-
+external int_of_bool : bool -> int = "%identity"
+external bool_of_int : int -> bool = "%identity"
```

Comment:
Elegant, but aren't we now operating on a bool that might technically be invalid (if the `int` wasn't `0` or `1`)?
At the call-site in `exists_uint8` this is guaranteed not to be the case due to the `select`, but I wonder if we could move this definition into that function to prevent accidental misuse?

Reply:
We can rename it `unsafe_bool_of_int` to prevent accidental misuse, yes :+1: